AtmosphereAutopilot/Modules/ProgradeThrustController.cs

```diff
@@ -40,6 +40,96 @@ public struct SpeedSetpoint
         public const float kts2mps = 0.514444f;
         public const float mps2kts = 1.0f / kts2mps;
 
+        public const float machEpsilon = 0.005f;
+
+        public static float rayleighPitotTubeStagPressureSubSonic(float M, float gamma)
+        {
+            float pressure;
+
+            pressure = (M * M) * (gamma - 1) / 2 + 1;
+            pressure = Mathf.Pow(pressure, gamma / (gamma - 1));
+
+            return pressure;
+        }
+
+        public static float rayleighPitotTubeStagPressureSuperSonic(float M, float gamma)
+        {
+            float pressure;
+
+            pressure = (gamma + 1) * M;
+            pressure *= pressure;
+            pressure /= (4 * gamma * M * M - 2 * (gamma - 1));
+            pressure = Mathf.Pow(pressure, gamma / (gamma - 1));
+            pressure *= (1 - gamma + 2 * gamma * M * M) / (gamma + 1);
+
+            return pressure;
+        }
+
+        public static float mps2Ias(float mps, Vessel v)
+        {
+            float M = (float)(mps / v.speedOfSound);
+            float gamma = (float) v.mainBody.atmosphereAdiabaticIndex;
+            float pressureRatio;
+            if (M <= 1)
+            {
+                pressureRatio = rayleighPitotTubeStagPressureSubSonic(M, gamma);
+            }
+            else
+            {
+                pressureRatio = rayleighPitotTubeStagPressureSuperSonic(M, gamma);
+            }
+
+            float velocity = pressureRatio - 1;
+            velocity *= (float)v.staticPressurekPa * 1000 * 2;
+            velocity /= 1.225f;
+            velocity = Mathf.Sqrt(velocity);
+            return velocity;
+        }
+
+        public static float rayleighMachFromPitotTubeStagPressure(float pressure, float gamma)
+        {
+            float M;
+
+            float soundPres = Mathf.Pow((gamma + 1) / 2, gamma / (gamma - 1)); ;
+
+            if (pressure <= soundPres)
+            {
+                M = Mathf.Pow(pressure, (gamma - 1) / gamma);
+                M = (M - 1) * 2 / (gamma - 1);
+                M = Mathf.Sqrt(M);
+            }
+            else
+            {
+                float slope, mnew;
+                float m1 = 1, m2 = 2;
+                float pt1 = rayleighPitotTubeStagPressureSuperSonic(m1, gamma) - pressure;
+                float pt2 = rayleighPitotTubeStagPressureSuperSonic(m2, gamma) - pressure;
+                do
+                {
+                    slope = (m2 - m1) / (pt2 - pt1);
+                    mnew = m2 - slope * pt2;
+                    mnew = Mathf.Max(1, mnew); // don't go subsonic
+                    m1 = m2;
+                    m2 = mnew;
+                    pt1 = pt2;
+                    pt2 = rayleighPitotTubeStagPressureSuperSonic(m2, gamma) - pressure;
+                } while (Mathf.Abs(m2 - m1) > machEpsilon);
```

Comment:
Let's put a limit to iteration count and print Error to log if we reach it.
Frankly, not a fan of such heavy math in our fixedUpdate =( 
Lot's of cycles for such a petty formularity.

Does the method always converge?
Can we replace it by some precalculated mesh interpolation?

Reply:
Sounds bold but I suggest using IAS alone for setpoint. Here's my 2 cents:
1. IRL aircrafts never knows their true speed against ground (and static air) hence IAS is their only choise to read & hold (along with mach at higher altitude, which is also calculated from IAS);
2. As said before, IAS plays an important role in aerodynamic and maneuverability;
3. All other speed modes in AA, can be converted to IAS with O(1) algorithms.

Reply:
1). I never intended to constrain AAP to what IRL aircraft knows. I used every easily-accessible information the game gave me.
2). ...in FAR. Maybe we should only use these formulas when FAR is installed? There already exists FAR-specific code in AAP, so that would be easy to code and definitely easy my mind.
3). Nah, let's keep it in kinematic speed. Simple is good.

Reply:
I came with another possible solution.
I've found a way to read IAS value directly from FAR so we may ditch the calculation code.
Then, I want to make another change: we determine the airspeed differencies from inside the Prograde Thrust Controller and ditch the need to call mps() method outside the class. So we can and apply thrust controls without the need of the IAS reverse functions. It will also make PTC a StateController rather than a SISOController.

If FAR is not found, we can then use the good old EAS code.

Reply:
Sounds good! If you feel comfortable with runtime reflections, that would be the simplest solution, IMO.